File: baekjoon/10989.py
# 수 정렬하기 3
# 정렬

#Counting Sort

# import sys
# N = int(sys.stdin.readline())
# arr = []
# max = 0
# for _ in range(N) :
#     inp = int(sys.stdin.readline())
#     arr.append(inp)
#     if inp > max :
#         max = inp
# count = [0]*max
# for i in arr :
#     count[i-1] += 1
# for i in range(len(count)) :
#     if i != 0 :
#         count[i] += count[i-1]
# answer = [None]*N
# for i in reversed(arr) :
#     answer[count[i-1]-1] = i
#     count[i-1] -= 1
# for i in answer :
#     print(i)

import sys
n = int(sys.stdin.readline())
b = [0] * 10001
for i in range(n):
    b[int(sys.stdin.readline())] += 1
for i in range(10001):
    if b[i] != 0:
        for j in range(b[i]):
            print(i)

#Python Sort

# import sys
# N = int(sys.stdin.readline())
# arr = []
# for _ in range(N) :
#     arr.append(int(sys.stdin.readline()))
# for i in sorted(arr) :
#     print(i)

File: baekjoon/10989.cpp
// 수 정렬하기 3
// 정렬

// 단순하게 저장받았다가 정렬하면 
// 8MB 메모리 초과가 뜸
// 따라서 각 수 별로 카운팅 해야됨

// #include <iostream>
// #include <algorithm>

// int N;
// int arr[10000000];

// int main(){

//     std::ios_base::sync_with_stdio(0);
//     std::cin.tie(0);

//     std::cin >> N;

//     for(int i = 0; i < N; i++){
//         std::cin >> arr[i];
//     }

//     std::sort(arr, arr + N);

//     for(int i = 0; i < N; i++){
//         std::cout << arr[i] << '\n';
//     }

//     return 0;
// }

#include <iostream>

int N, inp;
int cnt[10001];

int main(){

    std::ios_base::sync_with_stdio(0);
    std::cin.tie(0);

    std::cin >> N;

    for(int i = 0; i < N; i++){
        std::cin >> inp;
        cnt[inp]++;
    }

    for(int i = 1; i <= 10000; i++){
        while(cnt[i]--){
            std::cout << i << '\n';
        }
    }

    return 0;
}
